zerver/tests/test_invite.py

```diff
@@ -2607,6 +2717,27 @@ def test_multiuse_link_with_specified_streams(self) -> None:
             name5, [rome, default_streams[0], default_streams[1], default_streams[2]]
         )
 
+    def test_multiuse_link_with_specified_user_groups(self) -> None:
+        iago = self.example_user("iago")
+        self.login("iago")
+
+        user_group1 = create_user_group_in_database("test1", [], iago.realm, acting_user=iago)
+        user_group2 = create_user_group_in_database("test2", [], iago.realm, acting_user=iago)
+
+        user_groups = [user_group1, user_group2]
+
+        invite_link = self.generate_multiuse_invite_link(user_groups=user_groups)
+        self.check_user_able_to_register(self.nonreg_email("bob"), invite_link)
+
+        # bob is a direct member of two role-based system groups also.
+        user_groups_subscriptions = get_direct_user_groups(self.nonreg_user("bob"))
+        user_group_names = [group.named_user_group.name for group in user_groups_subscriptions]
+
+        self.assertEqual(
+            set(user_group_names),
+            {"test1", "test2", SystemGroups.MEMBERS, SystemGroups.FULL_MEMBERS},
+        )
```

Comment:
This test seems identical to test_create_multiuse_link_with_specified_user_groups_api_call? See https://zulip.readthedocs.io/en/stable/testing/philosophy.html#integration-testing-or-unit-testing for thinking about what we need.

Reply:
When writing these tests, I aimed to implement a structure similar to that used for streams, given their parallel nature. Streams exhibit identical functions, as demonstrated by `test_create_multiuse_link_with_specified_streams_api_call` and `test_successful_invite_users_with_specified_streams`.

Reply:
I have kept this test for now, but I guess we can remove these. Should we do a prep commit removing such other duplicate tests as well? I have not yet seen how that will affect coverage but that can be handled.

Reply:
That can be a follow-up; not urgent to do it, but best to not lose track of it, since duplicate tests can be extra maintenance work.